pkg/scheduler/internal/cache/node_tree_test.go

```diff
@@ -400,14 +400,14 @@ func TestNodeTreeMultiOperations(t *testing.T) {
 			nodesToAdd:     append(allNodes[4:9], allNodes[3]),
 			nodesToRemove:  nil,
 			operations:     []string{"add", "add", "add", "add", "add", "next", "next", "next", "next", "add", "next", "next", "next"},
-			expectedOutput: []string{"node-4", "node-5", "node-6", "node-7", "node-3", "node-8", "node-4"},
+			expectedOutput: []string{"node-4", "node-5", "node-6", "node-7", "node-4", "node-5", "node-6"},
```

Comment:
Hmm... This seems to be incorrect. We are not seeing node-8 and node-3 before seeing node-4 for the second time. We haven't changed the logic of node tree. Why this test is changed?

Reply:
We kind of changed the logic: every time a node is added or removed, recomputeAllNodes is invoked, which resets all indicies since it calls resetExhausted, which basically resets nt.next() to start from the beginning.

Reply:
You are right. I forgot about that. We should make sure that we have test cases that ensures all nodes appear in `next` output.

Reply:
Done.